### PR TITLE
soc: pm:  Microchip MEC172x SoC based power management

### PR DESCRIPTION
**Pull request by jvasanth1**
Add support for SoC power management for Microchip MEC172x.

### CODE REVIEW

File: soc/arm/microchip_mec/mec172x/device_power.c [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+	/* Enable PS2_0B_WK */
+	regs->GIRQ21.SRC = MCHP_PS2_0_PORT0B_WK_GIRQ_BIT;
+	regs->GIRQ21.EN_SET = MCHP_PS2_0_PORT0B_WK_GIRQ_BIT;
+#endif
+#endif
+}
+
+void soc_deep_sleep_wake_dis(void)
+{
+#ifdef CONFIG_PS2_XEC_0
```
Comment (albertofloyd) [inline soc/arm/microchip_mec/mec172x/device_power.c:112]: replace this with device tree checks as pointed in recent PS2 driver PR

File: soc/arm/microchip_mec/mec172x/device_power.c [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+void soc_deep_sleep_wake_en(void)
+{
+#if defined(CONFIG_KSCAN) || defined(CONFIG_PS2_XEC_0)
+	struct ecia_named_regs *regs = ECIA_XEC_REG_BASE;
+#if defined(CONFIG_KSCAN)
+	/* Enable PLL wake via KSCAN  */
+	regs->GIRQ21.SRC = MCHP_KEYSCAN_GIRQ_BIT;
+	regs->GIRQ21.EN_SET = MCHP_KEYSCAN_GIRQ_BIT;
+#endif
+#if defined(CONFIG_PS2_XEC_0)
```
Comment (albertofloyd) [inline soc/arm/microchip_mec/mec172x/device_power.c:102]: Use ifdef with DT_ macros for PS2 instances

File: soc/arm/microchip_mec/mec172x/device_power.c [hunk 3]
```diff
@@ -0,0 +1,384 @@
...
+	struct ecia_named_regs *regs = ECIA_XEC_REG_BASE;
+
+	regs->GIRQ22.SRC = UINT32_MAX;
+	regs->GIRQ22.EN_SET = MCHP_ESPI_WK_CLK_GIRQ_BIT;
+#endif
+}
+
+void soc_deep_sleep_non_wake_dis(void)
+{
+#ifdef CONFIG_ESPI_XEC
```
Comment (albertofloyd) [inline soc/arm/microchip_mec/mec172x/device_power.c:84]: Use either CONFIG_ESPI_XEC_V2 or simply CONFIG_ESPI

File: soc/arm/microchip_mec/mec172x/device_power.c [hunk 4]
```diff
@@ -0,0 +1,384 @@
...
+ * the EC. Once the peripheral has serviced the external master the PLL
+ * will be turned back off. For example, if the eSPI master requests eSPI
+ * configuration information or state of virtual wires the EC doesn't need
+ * to be involved. The hardware can power on the PLL long enough to service
+ * the request and then turn the PLL back off.  The SMBus and I2C peripherals
+ * in slave mode can also make use of this feature.
+ */
+void soc_deep_sleep_non_wake_en(void)
+{
+#ifdef CONFIG_ESPI_XEC
```
Comment (albertofloyd) [inline soc/arm/microchip_mec/mec172x/device_power.c:74]: Use CONFIG_ESPI

File: soc/arm/microchip_mec/mec172x/device_power.h [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+};
+
+struct ds_dev_info {
+	uint32_t ecs[2];
+	uint32_t timers[NUM_DS_TIMER_ENTRIES];
+#ifdef DEEP_SLEEP_UART_SAVE_RESTORE
+	uint8_t uart_info[MCHP_UART_INSTANCES];
+#endif
+	/* Analog power */
+#ifdef CONFIG_PECI_XEC
```
Comment (albertofloyd) [inline soc/arm/microchip_mec/mec172x/device_power.h:76]: Use CONFIG_PECI for consistency

File: soc/arm/microchip_mec/mec172x/device_power.h [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+	struct ds_peci_info peci_info;
+#endif
+#ifdef CONFIG_I2C_XEC_V2
```
Comment (albertofloyd) [inline soc/arm/microchip_mec/mec172x/device_power.h:79]: Use CONFIG_I2C

File: soc/arm/microchip_mec/mec172x/device_power.c (hunk 1 above)
Comment (jvasanth1) [reply soc/arm/microchip_mec/mec172x/device_power.c]: updated

File: soc/arm/microchip_mec/mec172x/device_power.c (hunk 2 above)
Comment (jvasanth1) [reply soc/arm/microchip_mec/mec172x/device_power.c]: updated

File: soc/arm/microchip_mec/mec172x/device_power.c (hunk 3 above)
Comment (jvasanth1) [reply soc/arm/microchip_mec/mec172x/device_power.c]: updated

File: soc/arm/microchip_mec/mec172x/device_power.h (hunk 5 above)
Comment (jvasanth1) [reply soc/arm/microchip_mec/mec172x/device_power.h]: updated

File: soc/arm/microchip_mec/mec172x/device_power.h (hunk 6 above)
Comment (jvasanth1) [reply soc/arm/microchip_mec/mec172x/device_power.h]: updated

**Review by albertofloyd — DISMISSED** [review]

**Comment by MaureenHelm** [comment]
Please fix the failed compliance checks

**Comment by albertofloyd** [comment]
Fixes https://github.com/zephyrproject-rtos/zephyr/issues/40972

**Comment by jvasanth1** [comment]
Hi @MaureenHelm fixed the compliance check errors.
Hi @albertofloyd please help to re-approve

**Review by albertofloyd — APPROVED** [review]

**Review by ceolin — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**